static/js/pages/applications/ApplicationDashboardPage.js

```diff
@@ -72,57 +78,277 @@ export class ApplicationDashboardPage extends React.Component<Props, State> {
+          ) : null}
         </div>
-        <div className="col-12">
-          <h3>Payment</h3>
+        <div className="col-12 col-sm-5 text-sm-right">
           <a
             className="btn-link"
-            onClick={() => {
-              setDrawerState(PAYMENT)
-              setDrawerMeta({
-                applicationId
-              })
-              setDrawerOpen(true)
-            }}
+            onClick={R.partial(openDrawer, [{ type: PROFILE_VIEW }])}
           >
-            Make a Payment
+            View/Edit Profile
           </a>
         </div>
+      </React.Fragment>,
+      "profile",
+      profileFulfilled
+    )
+
+    const resumeFulfilled = !!applicationDetail.resume_upload_date
+    const resumeRow = this.renderDetailRow(
+      <React.Fragment>
+        <h3>Resume or LinkedIn Profile</h3>
+        <div className="col-12 col-sm-6 status-text">
+          {applicationDetail.resume_upload_date && (
+            <div>
+              <span className="label">Completed: </span>
+              {formatReadableDateFromStr(applicationDetail.resume_upload_date)}
+            </div>
+          )}
+        </div>
+        {profileFulfilled && (
+          <div className="col-12 col-sm-5 text-sm-right">
+            <a className="btn-link">
+              {resumeFulfilled ?
+                "View/Edit Resume or LinkedIn Profile" :
+                "Add Resume or LinkedIn Profile"}
+            </a>
+          </div>
+        )}
+      </React.Fragment>,
+      "resume",
+      resumeFulfilled
+    )
+
+    let stepFulfilled, submissionApproved
+    const submissionStepRows = applicationDetail.run_application_steps.map(
+      (step: ApplicationRunStep) => {
+        // If this is the first required submission, the user should only be able
+        // to submit something if the resume step has been completed. Otherwise, they should
+        // only be able to submit something if their previous submission was approved.
+        const stepReady =
+          stepFulfilled === undefined ? resumeFulfilled : stepFulfilled
+        stepFulfilled = appStepSubmissions[step.id] !== undefined
+        submissionApproved =
+          !!appStepSubmissions[step.id] &&
+          appStepSubmissions[step.id].review_status === REVIEW_STATUS_APPROVED
+        return (
+          <React.Fragment key={step.step_order}>
+            {this.renderDetailRow(
+              this.SUBMISSION_STEP_RENDERERS[step.submission_type](
+                step,
+                appStepSubmissions[step.id],
+                stepFulfilled,
+                stepReady
+              ),
+              "submission",
+              stepFulfilled
+            )}
+            {this.renderDetailRow(
+              this.renderReviewStep(appStepSubmissions[step.id]),
+              "review",
+              submissionApproved
+            )}
+          </React.Fragment>
+        )
+      }
+    )
+
+    const paymentRow = this.renderDetailRow(
+      <React.Fragment>
```

Comment:
Passing a fragment into a function is a bit of an antipattern here, implementing `renderDetailRow` as an HOC and then wrapping this fragment (which should be its own component too) with that HOC as a concrete component will make this a lot cleaner and easier to reason about.